### PR TITLE
feat(cesium-map): viewer, layers + stories

### PR DESCRIPTION
**Pull request by alebinson**
Closes issues: 
#64 

## Checklist
- [ ] Tests
- [X] Stories
- [X] Lint
- [X] Prettier

## What I did

- [ ] BREAKING CHANGE
- [X] feature request
- [ ] bug fix
- [X] documentation

More information:
Cesium map component implemented to be integrated in our react apps.

### CODE REVIEW

**Comment by alebinson** [comment]
@syncush Default story changed by defining stories loading order in main.js.
Waiting to PR

File: packages/react-components/.gitignore [hunk 1]
```diff
@@ -22,4 +22,4 @@ yarn-debug.log*
 yarn-error.log*
 
 #storybook
-/storybook-static
+/storybook-static/cesium
```
Comment (syncush) [inline packages/react-components/.gitignore]: add in lower and upper case

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  const pixelDistance = geodesic.surfaceDistance;
+
+  // Find the first distance that makes the scale bar less than 100 pixels.
+  const maxBarWidth = 100;
+  let distance;
+  for (let i = distances.length - 1; !isNumber(distance) && i >= 0; --i) {
+    if (distances[i] / pixelDistance < maxBarWidth) {
+      distance = distances[i];
+    }
+  }
```
Comment (syncush) [inline packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx:112]: can we export the logic outside of the component? might be useful for other maps components

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        mapViewer,
+        scaleData,
+        setScaleData,
+        props.locale
+      );
+    };
+
+    const helper = new EventHelper();
+    const tileLoadHandler = (event: number): void => {
+      // console.log("Tiles to load: " + event, mapViewer.scene.globe.tilesLoaded);
```
Comment (syncush) [inline packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx]: remove

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    // see here https://community.cesium.com/t/globe-pick-returning-undefined/6616
+    helper.add(mapViewer.scene.globe.tileLoadProgressEvent, tileLoadHandler);
+
+    mapViewer.camera.moveEnd.addEventListener(setFromEvent);
+    // mapViewer.camera.changed.addEventListener(setFromEvent);
+
+    return (): void => {
+      try {
+        mapViewer.camera.moveEnd.removeEventListener(setFromEvent);
+        // mapViewer.camera.changed.removeEventListener(setFromEvent);
```
Comment (syncush) [inline packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx]: remove

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      {isNumber(scaleData.barWidth) && (
+        <>
+          <div className="distance-legend-label">{scaleData.distanceLabel}</div>
+          <div
+            className="distance-legend-scale-bar"
+            style={{
+              height: '2px',
+              width: `${scaleData.barWidth.toString()}px`,
+              // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+              left: `${(5 + (125 - scaleData.barWidth) / 2).toString()}px`,
```
Comment (syncush) [inline packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx]: I think we should extract the calculation from the jsx and let it sit in its own function

File: packages/react-components/src/lib/utils/map.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
```
Comment (syncush) [inline packages/react-components/src/lib/utils/map.ts:1]: I think you should not use magic numbers and export them to `consts`

**Review by syncush — CHANGES_REQUESTED** [review]

File: packages/react-components/.gitignore (hunk 1 above)
Comment (alebinson) [reply packages/react-components/.gitignore]: Done

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx (hunk 2 above)
Comment (alebinson) [reply packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx:112]: Think about it, it's too much depends on CESIUM. Not sure it should be as a generic one.
Let's stick with local and due to the needs perform appropriate refactoring(hate this word)

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx (hunk 3 above)
Comment (alebinson) [reply packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx]: Done

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx (hunk 4 above)
Comment (alebinson) [reply packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx]: Done

File: packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx (hunk 5 above)
Comment (alebinson) [reply packages/react-components/src/lib/cesium-map/tools/scale-tracker.tool.tsx]: Yep, U r right
Done

File: packages/react-components/src/lib/utils/map.ts (hunk 6 above)
Comment (alebinson) [reply packages/react-components/src/lib/utils/map.ts:1]: Basically you are right, but not in this case. The calculation is good documented, so extracting to const doesn't add more info.
It's very specific

**Review by syncush — APPROVED** [review]

**Pull request merged**